test/end-to-end/company.js

```diff
@@ -0,0 +1,212 @@
+            obj._id = obj._id.toString();
+            return obj;
+        });
+
+describe("Company application endpoint", () => {
+    describe("GET /company", () => {
+        beforeAll(async () => {
+            await Company.deleteMany({});
+        });
+
+        test("should return an empty array", async () => {
+            const res = await request()
+                .get("/company");
+            expect(res.status).toBe(HTTPStatus.OK);
+            expect(res.body.companies).toEqual([]);
+            expect(res.body.totalDocCount).toEqual(0);
+        });
+
+        test("should return the newly created company", async () => {
+            await Company.create({ name: "Company" });
+            const res = await request()
+                .get("/company");
+            const companies = await getCompanies({});
+            expect(res.status).toBe(HTTPStatus.OK);
+            expect(res.body.companies).toEqual(companies);
+            expect(res.body.totalDocCount).toEqual(1);
+        });
+    });
+
+    describe("POST /company/finish", () => {
+
+        describe("Without Auth", () => {
+            test("should respond with forbidden", async () => {
+                const emptyRes = await request()
+                    .post("/company/finish");
+
+                expect(emptyRes.status).toBe(HTTPStatus.UNAUTHORIZED);
+            });
+        });
+
+        describe("With Auth", () => {
+            const test_agent = agent();
+            const test_user = {
+                email: "user@email.com",
+                password: "password123",
+            };
+
+            const company_data = {
+                name: "Company Ltd"
+            };
+
+            beforeEach(async () => {
+                await Company.deleteMany({});
+                const c = await Company.create({ name: company_data.name });
+                await Account.deleteMany({});
+                await Account.create({ email: test_user.email, password: await hash(test_user.password), company: c._id });
+
+                // Login
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user)
+                    .expect(200);
+            });
+
+            test("should finish the application", async () => {
+                await test_agent
+                    .post("/company/finish")
+                    .attach("logo", "test/data/logo-niaefeup.png")
+                    .field("bio", "A very interesting and compelling bio")
+                    .field("contacts", ["contact1", "contact2"])
+                    .expect(HTTPStatus.OK);
+
+                const c = [... await Company.find({}).exec()][0];
+                expect([...c.contacts]).toEqual(["contact1", "contact2"]);
+                expect(c.finished).toBe(true);
+                expect(c.bio).toBe("A very interesting and compelling bio");
+                const filename = path.join(__dirname, `../../${c.logo.replace("static", "public")}`);
```

Comment:
Why is this replace needed? What's going on between static vs public?

Reply:
All files uploaded are saved to the `public` folder but the route to access the files is `static/file_path`, maybe i should change the name of one of them